amd_openvx_extensions/amd_rpp/source/tensor/Normalize.cpp

```diff
@@ -77,10 +88,40 @@ static vx_status VX_CALLBACK refreshNormalize(vx_node node, const vx_reference *
             dst_roi[i].xywhROI.roiHeight = src_roi[i].xywhROI.roiHeight;
         }
         data->pSrcRoi = static_cast<unsigned *>(data->pSrcDims);
+    } else if (numDims == 4) {
+        RpptROI *src_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr);
+        RpptROI *dst_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr_dst);
+        for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+            // rocAL ROI for image formats is stored in XYWH format. Transpose kernel needs ROI for all dims so adding the channel ROI here
+            unsigned index = i * nDim * 2;
+            if (data->inputLayout == vxTensorLayout::VX_NHWC) {
```

Comment:
I am confused about this data conversion? what is the difference between nDims and numDims?
Also NHWC means channel will be 0 index. So data->pSrcDims[index + 0] represents channel roi, isn't it?

Reply:
Also I think I mentioned this before. We shouldn't do data conversions here. vx_rpp has to be a thin layer which passes data to RPP without manipulating it

Reply:
Another confusion here is the way RPP handles the x,y,z indeces are reversed. For e.g, NHWC means [0] = c, [1] = H, [2]=W etc. Please discuss with Abishek why this is interpreted reversed format.